backend/routes/talkRouter.js

```diff
@@ -0,0 +1,102 @@
+var express = require('express');
+var passport = require('passport');
+var httpStatus = require('http-status-codes');
+var jwt = require('jwt-simple');
+var config = require('../config/main');
+var Talk = require('../models/talk');
+
+Array.prototype.sum = function (prop) {
+    var total = 0;
+    for (var i = 0; i < this.length; i++) {
+        total += this[i][prop]
+    }
+    return total;
+};
+
+var talkRouter = express.Router();
+
+var requireAuth = passport.authenticate('jwt', {session: false});
+
+talkRouter.route('/')
+    .all(requireAuth)
+    .get(function (req, res, next) {
+        var promise = Talk.find({})
+            .populate('postedBy', 'username')
+            .populate('ratings.ratedBy')
+            .exec(function(err, talks) {
+
+                if (err) throw err;
```

Comment:
Thanks comments. You are right. I missed the details of error handles. **next** should be better.